### ISSUE TITLE
is_admin() checks and AJAX context

### ISSUE THREAD

**Issue by felipeelia** [10up/ElasticPress#2148]
**Describe the bug**

As shared in #2114 by @moritzlang, several of our conditionals for `is_admin()` should take into account AJAX requests, where `is_admin()` is `true`.

We already have the `ep_ajax_wp_query_integration` filter that could (and should) be used in those tests. To keep things clean, we can create a new utils function called `is_admin_or_ajax()` that would contain both conditionals in it.

**Steps to Reproduce**

Run a WP_Query with `ep_integrate` in an AJAX request and see that even the weighting fields are included.

**Expected behavior**

Queries running via AJAX should mimic the behavior of the front end, especially when `ep_ajax_wp_query_integration` is true.

**Additional context**

Places we've identified that should use that new function:
- https://github.com/10up/ElasticPress/blob/3.5.6/includes/classes/Feature/Search/Weighting.php#L567
- https://github.com/10up/ElasticPress/blob/3.5.6/includes/classes/Feature/Autosuggest/Autosuggest.php#L224
Probably there are others.

**Comment by moritzlang** [comment]
Hi @felipeelia,

#### Regarding the issue description:
> **Steps to Reproduce**
> 
> Run a WP_Query with `ep_integrate` in an AJAX request and see that even the weighting fields are included.

Just to keep this correct: **no** weighting fields are included in this case. Also fuzziness is not changed!


#### I found another issue occurring regarding the `is_admin()` checks when `ep_ajax_wp_query_integration` is set to true (I need your help on this one):

Steps to Reproduce:
* `ep_ajax_wp_query_integration` is enabled
* Protected Content is (by default) disabled

Since `ep_ajax_wp_query_integration` evaluates to true, elasticpress will be used for **every** ajax query ([this is the related line in the code](https://github.com/10up/ElasticPress/blob/aa1d42fa202f804b74bec710dab46eceb8309353/includes/classes/Feature/Search/Search.php#L121)). Included ajax queries made by the wp backend (e.g. media library search). But because protected contents are not indexed, the search in the media library will return no results.

Solution:
We need to differentiate between ajax calls from the backend and ajax calls from the frontend [(is this a good way to deal with this? (see end of post)](https://florianbrinkmann.com/en/wordpress-backend-request-3815/). Additionally we have to take into account the "Protected Content" setting (How to evaluate if this is enabled? Is there a filter?), which enables elasticpress in the wp admin backend.

The condition in `Search.php` would look something like this:
```php
if ( ! ( $protected_content_enabled || ( ! is_admin_request() && $admin_integration ) ) ) {
  return;
}
```

Other `is_admin()` checks can then be removed, since the evaluation happens in `Search.php` and there should be no differentiation in other parts of the code.
I already incorporated a similar solution of this into my code base, which works fine.

**Comment by moritzlang** [comment]
Hi @felipeelia,

will this be fixed with the release of 3.6.0?

**Comment by lkraav** [comment]
Based on https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/load.php#L951 core convention seems to be:

```php
// Protect the admin backend.
if ( is_admin() && ! wp_doing_ajax() ) {
  return true;
}
```

**Comment by felipeelia** [comment]
Unfortunately, it wasn't @moritzlang . We've been actively working on it though, as it'll be needed to a feature we are creating but probably it'll be merged only for 3.7.0. Stay tuned!
And thanks for the tip, @lkraav!

**Comment by mckdemps** [comment]
@JakePT Do you have an update on this? Does this overlap with the work you're doing on the modal search?

**Comment by JakePT** [comment]
@mckdemps I did implement something for this for that feature. I'll look into how feasible it is to maybe pull it out and merge it separately.

**Comment by danielecelsa** [comment]
> Unfortunately, it wasn't @moritzlang . We've been actively working on it though, as it'll be needed to a feature we are creating but probably it'll be merged only for 3.7.0. Stay tuned!
> And thanks for the tip, @lkraav!

May I ask when you are planning to release version 3.7.0?

**Comment by felipeelia** [comment]
@danielecelsa the new function will be part of 3.6.2 planned for the next couple of days. Stay tuned!
